runelite-client/src/main/java/net/runelite/client/RuneLiteTerminator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018, Tomas Slusny <Adam@sigterm.info>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import net.runelite.client.discord.DiscordService;
+
+@Singleton
+public class RuneLiteTerminator implements AutoCloseable
+{
+	private final ClientSessionManager clientSessionManager;
+	private final DiscordService discordService;
+
+	@Inject
+	private RuneLiteTerminator(final ClientSessionManager clientSessionManager, final DiscordService discordService)
+	{
+		this.clientSessionManager = clientSessionManager;
+		this.discordService = discordService;
+	}
+
+	@Override
+	public void close()
+	{
+		clientSessionManager.shutdown();
+		discordService.close();
```

Comment:
Double check this really shuts down the client when it is closed since I recall this being an issue before.